### ISSUE TITLE
Get Python 3 installed in the docker images

### ISSUE THREAD

**Issue by cdrini** [internetarchive/openlibrary#3417]
Subtask of #3416

### Summary of Requirements

- [x] Should be able to run `python3.8 -m pytest` inside a docker image
- [x] The regular service should run using python 2
- [x] The infogami submodule should still be python 2 compatible
    - Idea: Maybe need 2 infogami submodules pinned to different versions?
- [ ] The tests _DO NOT_ need to all pass; but they need to give the same results as Travis on Python 3

### Stakeholders
@cclauss @paulproteus

**Comment by cclauss** [comment]
Which of these tasks is not already done in #3411 ?

**Comment by cdrini** [comment]
I need to test locally, to see if `pytest` works from the docker image. When I was playing around with that I think `pytest` needs infogami master to run, but need to test to make sure.

**Comment by cclauss** [comment]
Infogami master is brought in on [line 76](https://github.com/internetarchive/openlibrary/blob/20a84a92142d94d2ba664e8d426ac1413e21d482/docker/Dockerfile.olbase#L76).

**Comment by cdrini** [comment]
Ah, I think I was finding that the infogami master was causing errors on python2 😬 That's what the 3rd point is there. Let me test and see what happens :+1:

**Comment by cdrini** [comment]
Oh, misread. Infogami needs to be `master` in order for pytest to run with python 3...

**Comment by cclauss** [comment]
Yes.  #2947

**Comment by cdrini** [comment]
Ok, let's punt the infogami requirements to #3418 . I think getting a working python 3 is good enough for this step :+1:

**Comment by cdrini** [comment]
Or wait! I have an idea. We can create the python 3 flag now, and then run something like `PYTHON=3 docker-compose up -d` to launch the container with infogami pinned to master instead of what it currently is. That way we can run pytest locally using python 3, and then seamlessly switch to running the site with python2!

**Comment by cclauss** [comment]
Please no.  We want to be able to run the same mods on both Python 2 and Python 3 ___without rebooting___.  I can do this on my Mac (and I am sure that you can too) so why do we not want to support that in our proposed development environment?  I have not used Docker on this project because it is useless for A/B test which we really need now.

On Travis CI we have been doing `pushd vendor/infogami && git pull origin master && popd` for a long time now so why can't we use a similar pattern inside of Docker to switch Infogami when we need to?

**Comment by cdrini** [comment]
Rebooting the docker container takes like 1s on my machine; I do it like 1000 times a day. Note you don't have to _rebuild_ the image to run in python 3, just restart it. So here's the dev flow this would allow:

```sh
# Run tests with python2
docker-compose run --rm web make test

# Run tests with python3 
PYTHON3=true docker-compose run --rm web bash scripts/test-py3.sh
## After #3418, we should be able to just do
# PYTHON3=true docker-compose run --rm web make test

# Launch site with python2
docker-compose up -d

# Launch site with python3 (after #3418)
PYTHON3=true docker-compose up -d
```

And honestly, I couldn't think of way to make it happen inside docker without make the flow kind of awkward/requireing inside the docker image a bunch.

Can you give me an example of the flow you'd like to see/what you'd like to do?

**Comment by cclauss** [comment]
```
python2 -m pytest
pushd vendor/infogami && git pull origin master && popd
python3 -m pytest
```

**Comment by cdrini** [comment]
Well you can still do that:

```sh
$ docker-compose up -d
$ docker-compose exec web bash
web# make git
web# python2 -m pytest
web# pushd vendor/infogami && git pull origin master && popd
web# python3 -m pytest
```
(Note because the vendors directory persists between docker runs, you'll always have to run `make git` first to ensure it's using the prod version of infogami).

The `PYTHON3` flag is for launching the site with python 3. We will need such a flag for #3418 regardless, because the site is started automatically on `docker-compose up`. For quick a/b testing of the test files, you can do that flow :+1: But for a/b testing of the entire site, you do need to restart the container. Does that seem reasonable?
